app/assets/javascripts/pretty-text/addon/engines/discourse-markdown-it.js

```diff
@@ -58,6 +66,10 @@ function createHelper(
     pluginCallbacks.push([featureName, callback]);
   };
 
+  helper.requestCustomMarkdownCookFunction = (callback) => {
```

Comment:
`requestCustomMarkdownCookFunction` is a bit verbose for a public API. What about something like:

- buildCookFunction
- registerCook
- createCooker